### PR TITLE
Handle global dimensions

### PR DESCRIPTION
**Pull request by fredj**
- [x] WMTS support
- [x] WMS support (Image and Tile)

### CODE REVIEW

File: contribs/gmf/src/controllers/abstract.js [hunk 1]
```diff
@@ -123,6 +123,14 @@ gmf.AbstractController = function(config, $scope, $injector) {
   }];
 
   /**
+   * @type {Object.<string, string>}
+   * @export
+   */
+  this.dimensions = {
+    'Time': '20150211'
```
Comment (gberaudo) [inline contribs/gmf/src/controllers/abstract.js]: Is this hardcoded dimension needed?

**Comment by fredj** [comment]
@gberaudo ready to be reviewed, thanks

File: contribs/gmf/src/directives/layertree.js [hunk 2]
```diff
@@ -204,8 +207,69 @@ gmf.LayertreeController = function($http, $sce, $scope, ngeoCreatePopup,
...
+    }
+    if (!ol.object.isEmpty(dimensions)) {
+      var source = layer.getSource();
+      if (source instanceof ol.source.WMTS) {
+        source.updateDimensions(dimensions);
+      } else if (source instanceof ol.source.TileWMS || source instanceof ol.source.ImageWMS) {
+        source.updateParams(dimensions);
+      } else {
+        // the source is not ready yet
+        layer.once('change:source', function() {
```
Comment (gberaudo) [inline contribs/gmf/src/directives/layertree.js]: If `updateLayerDimensions_` is called multiple times before the `change:source` event is triggered. Won't it register many listeners?

File: contribs/gmf/src/directives/layertree.js [hunk 3]
```diff
@@ -74,6 +75,7 @@ ngeo.module.value('ngeoLayertreeTemplateUrl',
  *    possible also open a new window.
  *
  * @htmlAttribute {Object} gmf-layertree-source One theme (JSON).
+ * @htmlAttribute {Object} gmf-layertree-dimensions Global dimensions object.
```
Comment (gberaudo) [inline contribs/gmf/src/directives/layertree.js]: Shouldn't it be `Object.<string, string>`?

Comment (fredj) [inline contribs/gmf/src/directives/layertree.js]: yes, fixed

File: contribs/gmf/src/directives/layertree.js (hunk 2 above)
Comment (gberaudo) [inline contribs/gmf/src/directives/layertree.js]: OK, that is minor, let's merge this.

**Pull request merged**